class.phpmailer.php

```diff
@@ -1364,7 +1364,8 @@ public function postSend()
      */
     protected function sendmailSend($header, $body)
     {
-        if (!empty($this->Sender)) {
+        // CVE-2016-10033, CVE-2016-10045: Don't pass -f if characters will be escaped by escapeshellcmd when popen is called due to safe mode.
+        if (!empty($this->Sender) || ini_get('safe_mode')) {
```

Comment:
I think it should be `&& !ini_get('safe_mode')`. I checked that if `safe_mode` is not defined, e.g. because it's in PHP > 5.4, it behaves correctly.